In [3]:
import pandas as pd
from pathlib import Path

In [5]:
DATA_PATH = Path("../data/raw/higher_education.csv")

In [6]:
with open(DATA_PATH, "rb") as f:
    print(f.read(300))

b'ANIO;TIPO_UNIV;NIVEL_ACADEM;OF_ACADEM;DISCIP_OCDE;DISCIP_ESPECIF;TIPO_ALUMNO;U_MED;VALOR\r\n2014;Privado;Grado;Otros;Ciencias Agr\xedcolas;Ciencias Agropecuarias;EGRESADOS;Cantidad de personas;124\r\n2014;Privado;Grado;Otros;Ciencias Agr\xedcolas;Ciencias Agropecuarias;ESTUDIANTES;Cantidad de personas;1.475\r\n'


In [7]:
df = pd.read_csv(
    DATA_PATH,
    sep=";",
    encoding="latin-1"
)

In [8]:
df.head()

,ANIO,TIPO_UNIV,NIVEL_ACADEM,OF_ACADEM,DISCIP_OCDE,DISCIP_ESPECIF,TIPO_ALUMNO,U_MED,VALOR
0,2014,Privado,Grado,Otros,Ciencias Agrícolas,Ciencias Agropecuarias,EGRESADOS,Cantidad de personas,124.000
1,2014,Privado,Grado,Otros,Ciencias Agrícolas,Ciencias Agropecuarias,ESTUDIANTES,Cantidad de personas,1.475
2,2014,Privado,Grado,Otros,Ciencias Agrícolas,Veterinaria,EGRESADOS,Cantidad de personas,119.000
3,2014,Privado,Grado,Otros,Ciencias Agrícolas,Veterinaria,ESTUDIANTES,Cantidad de personas,2.134
4,2014,Privado,Posgrado,Doctorado,Ciencias Agrícolas,Ciencias Agropecuarias,EGRESADOS,Cantidad de personas,2.000


In [9]:
df = pd.read_csv(
    DATA_PATH,
    sep=";",
    encoding="latin-1",
    thousands="."
)

In [ ]:
df.info() #el dataset está completo y no hay valores nulos

<class 'pandas.DataFrame'>
RangeIndex: 6074 entries, 0 to 6073
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype
---  ------          --------------  -----
 0   ANIO            6074 non-null   int64
 1   TIPO_UNIV       6074 non-null   str  
 2   NIVEL_ACADEM    6074 non-null   str  
 3   OF_ACADEM       6074 non-null   str  
 4   DISCIP_OCDE     6074 non-null   str  
 5   DISCIP_ESPECIF  6074 non-null   str  
 6   TIPO_ALUMNO     6074 non-null   str  
 7   U_MED           6074 non-null   str  
 8   VALOR           6074 non-null   int64
dtypes: int64(2), str(7)
memory usage: 427.2 KB


In [11]:
df.describe()

,ANIO,VALOR
count,6074.000000,6074.000000
mean,2018.592690,3940.238064
std,2.903075,16632.086988
min,2014.000000,0.000000
25%,2016.000000,27.000000
50%,2019.000000,175.500000
75%,2021.000000,1116.750000
max,2023.000000,240214.000000


## Mientras se exploraba el dataset y cómo era interpretado, se encuentra lo siguiente:

- Separador de columnas: ";"
- Encoding: latin-1
- Separador de miles: "."
- Registros: 6074
- Columnas: 9
- Lapso temporal: 2014-2023